pkg/provisioner/ironic/ironic.go

```diff
@@ -686,6 +688,17 @@ func setDeployImage(config ironicConfig, accessDetails bmc.AccessDetails, hostIm
 	return nil
 }
 
+func redactedProbableURL(probableURL interface{}) interface{} {
+	if probableURL == nil || reflect.TypeOf(probableURL).Kind() != reflect.String {
```

Comment:
Why do you need reflection and interface{} here? a URL can only be a string.

Reply:
https://github.com/metal3-io/baremetal-operator/blob/58b41390fce5ab99896f5968f740dc442951b26c/pkg/provisioner/ironic/update_opts.go#L110 Here the input is an `interface{}`. I will change the input and return type to string and move the reflection logics to the function `sanitisedValue()`